Xplat/src/main/java/vazkii/botania/common/block/flower/generating/EntropinnyumBlockEntity.java

```diff
@@ -29,54 +27,77 @@
 import vazkii.botania.common.handler.BotaniaSounds;
 import vazkii.botania.xplat.XplatAbstractions;
 
+import java.util.ArrayList;
 import java.util.List;
 
 public class EntropinnyumBlockEntity extends GeneratingFlowerBlockEntity {
 	private static final int RANGE = 12;
 	private static final int EXPLODE_EFFECT_EVENT = 0;
 	private static final int ANGRY_EFFECT_EVENT = 1;
 
+	private static boolean trackTntEntities = false;
+	private static final List<PrimedTnt> trackedTntEntities = new ArrayList<>();
```

Comment:
Do we want to wrap this in a [ThreadLocal](https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/lang/ThreadLocal.html) to be defensive?

Reply:
I'm always wary of storing entities in global state like this. It makes memory leaks easy to slip in. How does `WeakHashMap<Level, IntOpenHashSet>` holding the entity id's sound, instead?